src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -244,6 +244,7 @@ public FetchCommand fetch_() {
             public boolean shallow;
             public Integer timeout;
             public boolean tags = true;
+            public Integer depth = 1;
```

Comment:
Upstream its always initialized to at least 1, so you would have to specify a negative number (which would also be invalid). If someone calls this code without that in mind and still sends a null through, an error will be thrown saying that null is not a valid depth (same as with negative numbers or 0). That error is thrown by the underlying git implementation and is stated clearly in the build log. If we want to disallow even saving the configuration with an invalid value, i'll definatly have to throw some more work in.